### PR TITLE
3420 upgrade flow test

### LINKED ISSUE #3420: Create a bridging test that can run on two different forks for fast- and slow-path

### LINKED ISSUE #3421: Create adding and removing liquidity fork test

### LINKED ISSUE #3422: Create adding and removing router liquidity fork test

### LINKED ISSUE #3424: Create a test to add and remove pools on upgrade

### PR DESCRIPTION
**Pull request by LayneHaber**
## Description

Fixes #3420 
Fixes #3421 
Fixes #3422 
Fixes #3424 

Based on #3486 

## Type of change

- [ ] Docs change / dependency upgrade
- [x] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## High-level change(s) description - from the user's perspective

Creates a forge tests that:

- add routers
- add router liquidity
- adds pool
- adds pool liquidity
- performs fast path transfer
- performs slow path transfer

on all combinations of supported destination / origin domains

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/deployments/contracts/contracts_forge/utils/MotherForker.sol [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+   * @notice Generate the diamond cut proposal by running `propose` via cli:
+   *     `ywc propose`
+   */
+  function utils_generateProposalFile() private {
+    string[] memory args = new string[](4);
+    args[0] = "yarn";
+    args[1] = "workspace";
+    args[2] = "@connext/smart-contracts";
+    args[3] = "propose";
+    vm.ffi(args);
```
Comment (rhlsthrm) [inline packages/deployments/contracts/contracts_forge/utils/MotherForker.sol:87]: Lol

**Review by rhlsthrm — APPROVED** [review]
LGTM

File: packages/deployments/contracts/contracts_forge/utils/MotherForker.sol [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+import "forge-std/console.sol";
+import "forge-std/StdJson.sol";
+
+/**
+ * @notice A 'ForkHelper' used to initialize all the applicable forks (foun`d in foundry.toml).
+ * @dev Should be inherited and used in the `upgrade` suite of tests to basically test upgradability
+ * of live contracts.
+ */
+abstract contract MotherForker is ForgeHelper {
```
[the remark this answers is not in the document]
Comment (just-a-node) [reply packages/deployments/contracts/contracts_forge/utils/MotherForker.sol:17]: Too good

[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/deployments/contracts/contracts_forge/utils/MotherForker.sol:17]: all my best work is in test files

**Comment by LayneHaber** [comment]
Testing on mainnets and found these two issues: #3687 #3686

**Pull request merged**